### PR TITLE
e2e: ginkgo timeouts: preparations

### PR DESCRIPTION
**Pull request by pohly**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

This contains several commits that are needed for https://github.com/kubernetes/kubernetes/pull/112923.

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

### CODE REVIEW

**Comment by aojea** [comment]
you have to fight with the linter Patric 😄 

/assign @spiffxp @aojea

**Comment by pohly** [comment]
The linter check is a problem: automatically injecting the context parameter into callbacks without any manual cleanup as done in the final commit of the full PR simply doesn't compile.

I could reorganize once more so that those two commits come last and not include them here, but I'm not sure whether it's worthwhile. Let's wait for @spiffxp to comment.

File: test/e2e/framework/internal/output/output.go [hunk 1]
```diff
@@ -124,8 +124,8 @@ func stripAddresses(in string) string {
 // locally) or one of a few relative paths (built in the Kubernetes CI).
 var stackLocation = regexp.MustCompile(`(?:/|vendor/|test/|GOROOT/).*/([[:^space:]]+.go:[[:digit:]]+)( \+0x[0-9a-fA-F]+)?`)
 
-// functionArgs matches "<function name>(...)".
-var functionArgs = regexp.MustCompile(`([[:alpha:]]+)\(.*\)`)
+// functionArgs matches "<function name>(...)" where <function name> may be an anoymous function (e.g. "pod_test.glob..func1.1")
```
Comment (mimowo) [inline test/e2e/framework/internal/output/output.go:127]:
```suggestion
// functionArgs matches "<function name>(...)" where <function name> may be an anonymous function (e.g. "pod_test.glob..func1.1")
```

**Comment by pohly** [comment]
Let's merge PR #112923 directly.

/close

**Pull request closed without merging**